mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -6603,9 +6605,8 @@ LogicalResult MaskOp::verify() {
                        "number of results");
 
   if (!llvm::equal(maskableOp->getResults(), terminator.getOperands()))
-    return emitOpError(
-        "expects all the results from the MaskableOpInterface to "
-        "be returned by the terminator");
+    return emitOpError("expects all the results from the MaskableOpInterface "
+                       "to match all the values returned by the terminator");
```

Comment:
After seeing the comment from @newling , I've realised that we are special-casing an empty `vector.mask`. This example will not trigger the error: 

```mlir
%0 = vector.mask %mask { vector.yield %a : vector<8xf32> } : vector<8xi1> -> vector<8xf32>
```

That's a bit of inconsistency. Perhaps leave a TODO to address this at some point?

Reply:
yes, my intent with this PR is to make clearer that an empty `vector.mask` is not a common valid case to mask operations and that may eventually go away. Let me clarify that a bit better in the doc. We would need the CSE equivalence issue to be fixed and improve some of the existing vector transformations. Definitely a target we should be moving towards.

Reply:
+1

I would still appreciate a TODO or some comment - mostly for my future self as a reminder about this conversation :)